browser/brave_rewards/test/creator_detection_browsertest.cc

```diff
@@ -0,0 +1,331 @@
+            __reactProps$123: { children: { props: { store: { getState } } } }
+          })
+        })
+      </script>
+    </head>
+    <body>
+      <div id="react-root">
+        <div></div>
+      </div>
+    </body>
+    </html>
+  )");
+
+  AddBlankTabAndShow(browser());
+  base::test::TestFuture<std::string> future;
+  TabHelperObserver observer(GetRewardsTabHelper(), future.GetCallback());
+  NavigateTo("twitter.com", "/testuser");
+
+  EXPECT_EQ(future.Get<0>(), "twitter#channel:987654321");
+}
+
+IN_PROC_BROWSER_TEST_F(CreatorDetectionBrowserTest, VimeoDetection) {
+  SetRewardsEnabled();
+
+  SetResponseHTML(R"(
+    <!doctype html>
+    <html>
+    <head>
+      <script type="application/ld+json">
+        [
+          {
+            "@type": "Person",
+            "identifier": "987654321",
+            "name": "Test User",
+            "url": "https://vimeo.com/testuser",
+            "image": "https://vimeo.com/user-avatar"
+          }
+        ]
+      </script>
+    </head>
+    <body>
+    </body>
+    </html>
+  )");
+
+  AddBlankTabAndShow(browser());
+  base::test::TestFuture<std::string> future;
+  TabHelperObserver observer(GetRewardsTabHelper(), future.GetCallback());
+  NavigateTo("vimeo.com", "/testuser");
+
+  EXPECT_EQ(future.Get<0>(), "vimeo#channel:987654321");
+}
+
+constexpr std::string_view kYouTubeHTML = R"(
+  <!doctype html>
+  <html>
+  <head>
+    <link rel="canonical" href="/channel/987654321" />
+  </head>
+  <body>
+    <ytd-video-owner-renderer>
+      <a href="/@testuser"></a>
+    </ytd-video-owner-renderer>
+    <div id="avatar">
+      <img src="/user-avatar" />
+    </div>
+  </body>
+  </html>
+)";
+
+IN_PROC_BROWSER_TEST_F(CreatorDetectionBrowserTest, YouTubeDetection) {
+  SetRewardsEnabled();
+  SetResponseHTML(kYouTubeHTML);
+
+  AddBlankTabAndShow(browser());
+  base::test::TestFuture<std::string> future;
+  TabHelperObserver observer(GetRewardsTabHelper(), future.GetCallback());
+  NavigateTo("www.youtube.com", "/@testuser");
+
+  EXPECT_EQ(future.Get<0>(), "youtube#channel:987654321");
+}
+
+IN_PROC_BROWSER_TEST_F(CreatorDetectionBrowserTest, InvalidHost) {
+  SetRewardsEnabled();
+  SetResponseHTML(kYouTubeHTML);
+  AddBlankTabAndShow(browser());
+
+  base::RunLoop run_loop;
+  base::OneShotTimer timeout;
+  timeout.Start(FROM_HERE, TestTimeouts::action_timeout(),
+                run_loop.QuitClosure());
+
+  NavigateTo("abc.youtube.com", "/@testuser");
+  EXPECT_EQ(GetRewardsTabHelper().GetPublisherIdForTab(), "");
+}
+
+IN_PROC_BROWSER_TEST_F(CreatorDetectionBrowserTest, RewardsDisabled) {
+  SetResponseHTML(kYouTubeHTML);
+  AddBlankTabAndShow(browser());
+
+  base::RunLoop run_loop;
+  base::OneShotTimer timeout;
+  timeout.Start(FROM_HERE, TestTimeouts::action_timeout(),
+                run_loop.QuitClosure());
+
+  NavigateTo("www.youtube.com", "/@testuser");
+  EXPECT_EQ(GetRewardsTabHelper().GetPublisherIdForTab(), "");
+}
```

Comment:
worth adding a test that nothing is processed if rewards is enabled but in an incognito tab?